packages/lib/models/Tag.test.js

```diff
@@ -132,28 +173,109 @@ describe('models/Tag', function() {
 		commonTags = await Tag.commonTagsByNoteIds([]);
 		expect(commonTags.length).toBe(0);
 
-		commonTags = await Tag.commonTagsByNoteIds([note0.id, note1.id, note2.id, note3.id]);
-		let commonTagIds = commonTags.map(t => t.id);
+		commonTags = await Tag.commonTagsByNoteIds([
+			note0.id,
+			note1.id,
+			note2.id,
+			note3.id,
+		]);
+		let commonTagIds = commonTags.map((t) => t.id);
 		expect(commonTagIds.length).toBe(0);
 
 		commonTags = await Tag.commonTagsByNoteIds([note1.id, note2.id, note3.id]);
-		commonTagIds = commonTags.map(t => t.id);
+		commonTagIds = commonTags.map((t) => t.id);
 		expect(commonTagIds.length).toBe(1);
 		expect(commonTagIds.includes(taga.id)).toBe(true);
 
 		commonTags = await Tag.commonTagsByNoteIds([note2.id, note3.id]);
-		commonTagIds = commonTags.map(t => t.id);
+		commonTagIds = commonTags.map((t) => t.id);
 		expect(commonTagIds.length).toBe(2);
 		expect(commonTagIds.includes(taga.id)).toBe(true);
 		expect(commonTagIds.includes(tagb.id)).toBe(true);
 
 		commonTags = await Tag.commonTagsByNoteIds([note3.id]);
 
-		commonTagIds = commonTags.map(t => t.id);
+		commonTagIds = commonTags.map((t) => t.id);
 		expect(commonTags.length).toBe(3);
 		expect(commonTagIds.includes(taga.id)).toBe(true);
 		expect(commonTagIds.includes(tagb.id)).toBe(true);
 		expect(commonTagIds.includes(tagc.id)).toBe(true);
-	}));
+	});
 
+	it('should sort tags in promptDialog', async () => {
+		const folder1 = await Folder.save({ title: 'folder1' });
```

Comment:
This test does not cover your changes at all.
Sure you copied the logic from `setTags` here but now if someone will accidentally breaks `setTags`, your test will still be passing. Which defeats the whole purpose of it.

Reply:
Then how do I test it? I will have to first export it using global variables, right? I am actually writing my first test here so just asking for feedback?

Reply:
Your test needs to call `setTags`

Reply:
I will write a new test accordingly, I just need a little push, is exporting a variable from a file for testing allowed? I don't see another way to test without exporting?

Reply:
But won't that test the entire file. I mean I just want to test the startTags and tagSuggestions from the entire file.

Reply:
If you don't want to test the whole file (which would be useful btw) you can extract the parts you want to test to separate methods and test them.

Reply:
Okay I will try to test the entire file. Also, extracting the parts using methods will work on classes but setTags uses functions. I researched about it on stackoverflow and found out that in order to export a variable from a function we will have to use global variables.